packages/web-components/fast-components/src/radio/radio.ts

```diff
@@ -0,0 +1,212 @@
+     * Passed down to proxy
+     */
+    @attr({ attribute: "checked", mode: "boolean" })
+    public checkedAttribute: boolean;
+    private checkedAttributeChanged(): void {
+        this.defaultChecked = this.checkedAttribute;
+    }
+
+    /**
+     * Initialized to the value of the checked attribute. Can be changed independently of the "checked" attribute,
+     * but changing the "checked" attribute always additionally sets this value.
+     */
+    @observable
+    public defaultChecked: boolean = !!this.checkedAttribute;
+    private defaultCheckedChanged(): void {
+        if (!this.dirtyChecked) {
+            // Setting this.checked will cause us to enter a dirty state,
+            // but if we are clean when defaultChecked is changed, we want to stay
+            // in a clean state, so reset this.dirtyChecked
+            this.checked = this.defaultChecked;
+            this.dirtyChecked = false;
+        }
+    }
+
+    /**
+     * The checked state of the control
+     */
+    @observable
+    public checked: boolean = this.defaultChecked;
+    private checkedChanged(): void {
+        if (!this.dirtyChecked) {
+            this.dirtyChecked = true;
+        }
+
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.checked = this.checked;
+        }
+
+        this.dispatchEvent(new CustomEvent("change", { bubbles: true, composed: true }));
+        this.checked ? this.classList.add("checked") : this.classList.remove("checked");
+        this.checkedAttribute = this.checked;
+
+        if (this.checked) {
+            this.updateOtherGroupRadios();
+        }
+    }
+
+    private updateOtherGroupRadios(): void {
+        if (this.name !== undefined && this.parentNode) {
+            const radioGroup = radioGroups.get(this.name);
+            radioGroup?.forEach((radio: FASTRadio) => {
+                if (radio.getAttribute("value") !== this.value) {
+                    radio.removeAttribute("checked");
+                    radio.checked = false;
+                }
+            });
+        }
+    }
+
+    protected proxy: HTMLInputElement = document.createElement("input");
+
+    /**
+     * Tracks whether the "checked" property has been changed.
+     * This is necessary to provide consistent behavior with
+     * normal input radios
+     */
+    // TODO: marjon check if this is needed for radio buttons?
+    private dirtyChecked: boolean = false;
+
+    constructor() {
+        super();
+        this.proxy.setAttribute("type", "radio");
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.updateForm();
+        if (this.name !== undefined) {
+            const radioGroup = radioGroups.get(this.name);
+            if (radioGroup) {
+                radioGroup.push(this);
+            } else {
+                radioGroups.set(this.name, [this]);
+            }
+        }
+
+        this.addEventListener("keydown", this.keydownHandler);
```

Comment:
I've been thinking about event bindings that are internal to elements lately...and I think you can move this line to the constructor and then completely remove the removeEventListener call in disconnect. I don't think there's a need to tear it down since the handler and the element are part of the single unit that will be GC'd.

Reply:
ok will do